packages/loader/container-loader/src/container.ts

```diff
@@ -740,9 +743,7 @@ export class Container
 		// Prefix all events in this file with container-loader
 		this.mc = loggerToMonitoringContext(ChildLogger.create(this.subLogger, "Container"));
 
-		this.options = {
-			...this.loader.services.options,
-		};
+		this.options = cloneDeep(this.loader.services.options);
```

Comment:
This is the bug fix.

Reply:
Is it correct to clone-deep the whole options (as opposed to just the client)?  I'm not sure if anything might depend on mutating a shared options object?

Reply:
nothing should depend on sharing of objects in the options object. i also believe we generally assume options are immutable, hence the pre-exiting shallow clone. this change looks fine to me.